backend/src/Models/VoterRolls.ts

```diff
@@ -0,0 +1,125 @@
+import { VoterRoll } from '../../../domain_model/VoterRoll';
+var format = require('pg-format');
+class VoterRollDB {
+
+    _postgresClient;
+    _tableName: string;
+
+    constructor(client: any, tableName: string) {
+        this._postgresClient = client;
+        this._tableName = tableName;
+    }
+
+    init(): Promise<VoterRollDB> {
+        console.log("VoterRollDB.init");
+        var query = `
+        CREATE TABLE IF NOT EXISTS ${this._tableName} (
+            voter_roll_id  SERIAL PRIMARY KEY,
+            election_id     VARCHAR,
+            voter_id        VARCHAR,
+            ballot_id       INTEGER,
```

Comment:
Could we remove submitted, and instead let ballot_id=null signify a non-submitted ballot? Not sure if that's a good idea or not, mostly curious if that's an option

Reply:
The only reason I'd say no is that I'm not sure if ballot_id should be here in the first place. As we discussed in the last meeting we may or may not want to be able to retrieve ballots for a specified user. I decided to add it for now.

Reply:
Cool, that makes sense

On a separate note, election_id, and voter_id kind of feel like keys in this table. There can be multiple entries per election or per user, but each election_id - voter_id pair should only appear once

Maybe we could use [composite keys ](https://www.simplilearn.com/tutorials/sql-tutorial/composite-key-in-sql#:~:text=A%20composite%20key%20in%20SQL%20can%20be%20defined%20as%20a,can%20uniquely%20identify%20any%20record.) here? I haven't used them before, but it could be a good fit, and it would allow us to remove voter_roll_id

Reply:
I didn't know about this, I was going to add code to check if the combination already exists but this is much simpler thanks! I looked up a postgres example and added it.

Reply:
Awesome, I'm glad it worked :D